xorg-server: update to 21.1.21

MingcongBai opened the pull request:
Topic Description
-----------------

- xorg-server: update to 21.1.21
    Co-authored-by: 白铭骢 \(Mingcong Bai\) \(@MingcongBai\) <jeffbai@aosc.io>

Package(s) Affected
-------------------

- xorg-server: 21.1.21

Security Update?
----------------

No

Build Order
-----------

```
#buildit xorg-server cx4-linux-graphics-driver-dri
```

Test Build(s) Done
------------------

**Primary Architectures**

- [ ] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`
- [x] LoongArch 64-bit (No SIMD) `loongarch64_nosimd`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

aosc-buildit-bot committed:
xorg-server: update to 21.1.21

Co-authored-by: 白铭骢 (Mingcong Bai) (@MingcongBai) <jeffbai@aosc.io>

xry111 commented:
```
Receiving objects:   2% (3259/120879), 1.04 MiB | 9.00 KiB/s 
error: RPC failed; curl 56 Recv failure: Connection reset by peer
error: 26070 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
```

:(

Cyanoxygen commented:
Confirmed that hybrid graphics with iGPU/NVIDIA works just fine. No additional configuration needed to use the NVIDIA GPU after disabling hybrid graphics.

MingcongBai commented:
> Confirmed that hybrid graphics with iGPU/NVIDIA works just fine. No additional configuration needed to use the NVIDIA GPU after disabling hybrid graphics.

Sounds good, I'll still need to submit a fix for cx4 and possibly LoongGPU.

MingcongBai commented:
> > Confirmed that hybrid graphics with iGPU/NVIDIA works just fine. No additional configuration needed to use the NVIDIA GPU after disabling hybrid graphics.
> 
> Sounds good, I'll still need to submit a fix for cx4 and possibly LoongGPU.

They both tested good, according to @xry111 and I.

MingcongBai committed:
cx4-linux-graphics-driver-dri: re-introduce X11 option GlxVendorLibrary

X.Org Server 21.1.21 addresses a previously introduced regression that
caused the driver to crash with GlxVendorLibrary specified (and
coincidentally made it redundant, as it internalised this option, which at
the same time the driver also implements).

Re-introduce this option now that X.Org Server no longer supplies it to
keep GLX working.
